initialize docs app

mpeterdev opened the pull request.

andy-haynes reviewed (changes requested):
Looks cool! Can you please remove the boilerplate/fake blog stuff?

Also I didn't see where the port is configured when it runs locally on 3000, but can we use something that isn't used by another dev service please?

mpeterdev commented:
@andy-haynes this is ready to go again. Initially I submitted the PR very quickly after initializing docusaurus in order to to get it on `main` and have a test Vercel deployment. You were right to withhold approval, I moved a little too fast there 😅 

I have removed almost all the default content, changed the port, and added some initial documentation

There are still some placeholder elements like logos, favicon, etc. We will get to those separately
